main.js

```diff
@@ -1,117 +1,117 @@
+    } else {
+        setValidationError("errorcheckData","Password is not correct");
+        showError("password_error",validationErrors.errorcheckData);
+    }
+    return false;
+}
 
+function checkUsername(){
+    if(username.value === userInfo.username ){
+        return true;   
+    } else {
+        setValidationError("errorcheckData","Username is not correct");
+        showError("username_error",validationErrors.errorcheckData);
+    }
     return false;
 }
 
-function fieldsClear() {
-    usernameInput.value = "";
-    userPassword.value = "";
+function setValidationError(fieldNameKey, error){
+    validationErrors[fieldNameKey] = error;
 }
 
-// Output exeption message
-function showError(elementId, textError)
-{    
-    removeError();
-    const error = document.getElementById(elementId).appendChild(document.createElement("p"));
-    error.classList.add("error");
-    error.textContent = textError;
+function showError(elementId, textError){
+    document.getElementById(elementId).textContent = textError;
 }
 
-// Delete exeption message
-function removeError(){
-    if(document.getElementById("form_username").contains(document.querySelector(".error"))){
-        document.getElementById("form_username").removeChild(document.querySelector(".error"));
-    } else if(document.getElementById("form_password").contains(document.querySelector(".error"))){
-        document.getElementById("form_password").removeChild(document.querySelector(".error"));
-    }
+function setFormValue(fieldNameKey, value){
+    formValues[fieldNameKey] = value;
 }
 
-// Creating html registration form
 function createLoginForm() {
     document.body.append(createFormContainer());
 
     const parentElement = document.getElementById("login");
     parentElement.appendChild(createTitle("Login form"));
-    parentElement.appendChild(createInputField("form_username","username", "Username"));
-    parentElement.appendChild(createInputField("form_password","password", "Password"));
+    parentElement.appendChild(createInputField({divId: "form_username", paragraphId: "username_error", paragraphClass: "error",
+        inputName: "username", labelText: "Username", inputValue: formValues.valueUsername, error: validationErrors.errorUsername}));
+    parentElement.appendChild(createInputField({divId: "form_password", paragraphId: "password_error", paragraphClass: "error",
+        inputName: "password", labelText: "Password", inputValue: formValues.valuePassword, error: validationErrors.errorPassword}));
     parentElement.appendChild(createButton("login_btn")); 
 }
 
 function createFormContainer(){
     const divFormContainer = createBaseElement({className: "form_container"});
-    divFormContainer.appendChild(createBaseElement({tagName: "form", className: "form", tagId: "login"}));
+    const loginForm = divFormContainer.appendChild(createBaseElement({tagName: "form", className: "form", tagId: "login"}));
+    
+    loginForm.addEventListener("submit",function(e){
```

Comment:
Move this function into arguments. `createFormContainer` isn't _**pure**_ now.
```
    function(e){
        e.preventDefault();
    
        if (isValidUsername() & isValidPassword()) {
            if (checkUsername() & checkPassword()) {
                userInfo.token = "xxx"
                alert("correct");
            }
        }  
    }
```

Reply:
Done